### ISSUE TITLE
[BUG] docker endpoint for "default" not found

### ISSUE THREAD

**Issue by milon27** [docker/compose#9956]
### Description

when building Dockefile with docker compose
```
docker compose build
```

got this error **docker endpoint for "default" not found** but if i try to build with a remote docker image compose works fine.

github repo:  https://github.com/milon27/hello-docker.git

### Steps To Reproduce

1. clone this repo  https://github.com/milon27/hello-docker.git
2. run ```docker compose up```
3. you should get **docker endpoint for "default" not found this error**

os: windows 10
docker desktop version: 4.13.0

note: it was fine when I update the docker desktop to the latest version this problem started.

### Compose Version

```Text
Docker Compose version v2.12.0
```

### Docker Environment

```Text
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.9.1)
  compose: Docker Compose (Docker Inc., v2.12.0)
  dev: Docker Dev Environments (Docker Inc., v0.0.3)
  extension: Manages Docker extensions (Docker Inc., v0.2.13)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.21.0)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 0
 Server Version: 20.10.20
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.16.3-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 6
 Total Memory: 12.39GiB
 Name: docker-desktop
 ID: P4Q6:4LDD:2OTW:O7QR:GYH5:4O4W:BHM6:XK5R:33DQ:AYA5:SMX7:WBHK
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
```

### Anything else?

Error screenshot

![image](https://user-images.githubusercontent.com/44096479/198014333-8f738965-2d65-4645-a525-c4d03574e367.png)

**Comment by milon27** [comment]
solved by: reinstalling docker desktop version 4.12.0

**Comment by pcartwright81** [comment]
Uninstalling and reinstalling didn't work for me.  Also tried removing & installing Hyper-V and WSL which also didn't work.

**Comment by Pixelao** [comment]
Have exactly the same problem, when trying to launch a project with docker-compose, trigger the message:

 **docker endpoint for "default" not found**

- Docker Engine: v20.10.20
- Docker Desktop: v4.13.0 (89412)
- Windows 11 Pro (21H2 22000.1165)

**Comment by llslim** [comment]
I am  having the same issue after installing docker desktop 4.13.0 on windows

**Comment by milon27** [comment]
solved by: reinstalling docker desktop version 4.12.0

**Comment by nelsonjchen** [comment]
`docker context ls` returns:

`invalid character '\x00' looking for beginning of value`

Traced it up to `~/.docker/contexts/meta/(some sha256)/meta.json`. `meta.json` was just full of `NUL`s. Deleted it, restarted Docker, and Docker 4.13.0 started working again.

So, a workaround, but bug wasn't solved.

**Comment by llslim** [comment]
Using @nelsonjchen workaround as a guide to my own solution.

On my Windows 10 machine machine, 'docker context ls' returns: 'too many redirects'. 

the ~/ .docker/contexts directory is non-existent. 

I renamed the ~/.docker directory to something elsse (i.e. ~/.docker-old) and restarted docker. 

Everything works.

**Comment by nelsonjchen** [comment]
I should clarify that the paths I mentioned are in WSL2's environment. They symlink to the "`C:`" or `/mnt` pathes.

**Comment by pcartwright81** [comment]
My `~/.docker/contexts/meta/(some sha256)/meta.json` was also full of nulls.  Deleted it, and restarted the docker service only.

**Comment by nelsonjchen** [comment]
> My `~/.docker/contexts/meta/(some sha256)/meta.json` was also full of nulls. Deleted it, and restarted the docker service only.

Missing the punchline. Did it work?

**Comment by pcartwright81** [comment]
Yes, updated comment.

**Comment by enjibby** [comment]
@nelsonjchen your workaround is a lifesaver. Thanks!

**Comment by milas** [comment]
Thanks for the reports - this looks like an issue in Docker Desktop itself (Compose does not handle management of contexts), so I'm closing in favor of docker/for-win#12561.

**Comment by asghar-mansourian** [comment]
@nelsonjchen thank you for help

**Comment by sdbondt** [comment]
> 

Just started using Docker and where what how do I find the .docker/contexts files?

**Comment by pcartwright81** [comment]
> > 
> 
> Just started using Docker and where what how do I find the .docker/contexts files?

%userprofile%\.docker\contexts

**Comment by wasuaje** [comment]
> docker context ls

This worked for me!

**Comment by AkhilGupta1983** [comment]
> `docker context ls` returns:
> 
> `invalid character '\x00' looking for beginning of value`
> 
> Traced it up to `~/.docker/contexts/meta/(some sha256)/meta.json`. `meta.json` was just full of `NUL`s. Deleted it, restarted Docker, and Docker 4.13.0 started working again. `meta.json` now has JSON.
> 
> Can build again.
> 
> So, a workaround, but bug wasn't exactly solved.

Thanks. This worked for me as well

**Comment by vibhordubey333** [comment]
For Windows Users

1. Delete `.docker` directory. Which exists on PATH `C:\Users\your-username\.docker`
2. Restart docker.

**Comment by iamThiagoo** [comment]
> For Windows Users
> 
> 1. Delete `.docker` directory. Which exists on PATH `C:\Users\your-username\.docker`
> 2. Restart docker.

Thanks man, it's works to me!

**Comment by rreis27** [comment]
I've tried deleting the .docker directory and restart docker, however it doesn't work and hasn't worked since last year sometime.  I'm running the following:

docker desktop: 4.16.2 (95914)
windows 10 Pro: 22H2

**Comment by maks-rafalko** [comment]
Since I didn't want to remove the whole `~/.docker` folder, I removed only `~/.docker/contexts/meta/(some sha256)/meta.json`. Solved the issue.

**Comment by TamilSekar12** [comment]
> Since I didn't want to remove the whole `~/.docker` folder, I removed only `~/.docker/contexts/meta/(some sha256)/meta.json`. Solved the issue.

@maks-rafalko Solution worked for me. Thanks

**Comment by Eji4h** [comment]
> Since I didn't want to remove the whole `~/.docker` folder, I removed only `~/.docker/contexts/meta/(some sha256)/meta.json`. Solved the issue.

Thank for you solution. It worked.

**Comment by RealMrHex** [comment]
still not fixed!
Docker Desktop 4.16.2 (95914)
Windows 11 22H2

**Comment by dionisvl** [comment]
up

**Comment by yilmazdurmaz** [comment]
Hi @milon27, Your post's title attracts attention, but the discussion seems to get longer over time. I suggest you add a TLDR at the top of your first post. You may compose your own or use this:

"""
TL;DR:
It turns out this error is caused by one or more broken docker "contexts" after an update, which is tracked in another issue.
A quick solution is to remove these broken contexts that reside in `%userprofile%\.docker\contexts`.
be careful as this includes "deleting" things from your computer.
"""

**Comment by ufedev** [comment]
![dockerConfig](https://user-images.githubusercontent.com/8621688/216231828-1c6f8a1a-b155-478f-a0c7-19fe1ce4c2a0.jpg)

disable Docker Compose V2that worked for me

**Comment by amrohendawi** [comment]
Resetting docker to factory defaults solves it for me.

**Comment by malkia** [comment]
Confirmed for me too - had a very old file in %USERPROFILE%\.docker\context\<some-sha>\meta.json which was zero bytes, as soon as I've deleted it, `docker buildx ls` shows now builders!

here is my info: (snipped some stuff at the end)

```
c:\> docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.10.0)
  compose: Docker Compose (Docker Inc., v2.15.1)
  dev: Docker Dev Environments (Docker Inc., v0.0.5)
  extension: Manages Docker extensions (Docker Inc., v0.2.17)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.23.0)
...
```

**Comment by soondook** [comment]
> ```powershell
> rm -r -force ~\.docker\contexts\meta\
> ```
Thank you! Your solution works!
I ran this command and then reloaded Docker Desktop!
docker desktop: 4.16.2 (95914)
windows 10 Pro: 22H2

**Comment by qiuyuhz** [comment]
> For Windows Users
> 
> 1. Delete `.docker` directory. Which exists on PATH `C:\Users\your-username\.docker`
> 2. Restart docker.

thanks,It worked for me !

**Comment by yilmazdurmaz** [comment]
People will, someday, learn to use that emoji button to tell us it works without telling us it works.

**Comment by Zetanova** [comment]
I got this situation multiple times in a row.
It looks like that the meta.json 0KB issue is only a secondary error and is produced when docker can't start fully up.

I believe one of the images got corrupted and simply made a break/timeout in multiple internal docker-host components  

after clearing the full image store twice it works gain.

**Comment by barguti** [comment]
> 

This worked for me (version 4.16.3), thanks @ufedev

**Comment by milas** [comment]
We are aware this is affecting a lot of Compose + Docker Desktop users so are reopening this issue for tracking and visibility.

Internally, we are working on this and will provide an update with the resolution once available. Thanks for your continued patience!

Thanks to everyone who has posted workarounds so far. If one is working for you, please use the GitHub 👍  reaction on the comment to leave a signal for others.

**Comment by QuAzI** [comment]
Docker Desktop 4.16.3 (96739)
Each time when I want to build app in Linux env I have to `rm -rf ~\.docker\contexts\meta\` because
```
$ docker context ls
unexpected end of JSON input
```

**Comment by ben-at-soul-refiner** [comment]
``` Bash
docker endpoint for "default" not found
```
was resolved by deleting `~/.docker/contexts/meta/(some sha256)/meta.json` as @nelsonjchen suggested

**Comment by yilmazdurmaz** [comment]
is it possible to lock the thread, tell people to read workaround solutions given countless times, and emphasize to **"open a new report only if none of them works"**?

or else, there will be thousands of "thanks, it works" messages in no time.

**Comment by QuAzI** [comment]
Technically none of them works for me because it is a pain to do it every 5
minutes

ср, 8 февр. 2023 г., 18:11 Yılmaz Durmaz ***@***.***>:

**Comment by yilmazdurmaz** [comment]
> "it is a pain to do it every 5 minutes".

Hold on there! You are the first one using this narrow interval among the posts I have seen. If you are not kidding around to fetch attention, your story can help identify the problem faster. Why don't you give it a try and spend some time:
- install 1-2 version older, and see if the problem persists. (dependence on a version)
- update again, and see if the problem persists. (dependence on a version)
- repeatable steps on how you build/run images, and run from compose files (dependence on configuration)
- a new bug post if your findings are long (to give more than awareness of the bug)

**Comment by QuAzI** [comment]
```
$ ls -l ~/.docker/contexts/meta/fe9c6bd7a66301f49ca9b6a70b217107cd1284598bfc254700c989b916da791e/meta.json
-rwxrwxrwx 1 quazi quazi 0 Feb  8 19:42 /home/quazi/.docker/contexts/meta/fe9c6bd7a66301f49ca9b6a70b217107cd1284598bfc254700c989b916da791e/meta.json

$ docker network ls
NETWORK ID     NAME      DRIVER    SCOPE
615c37f018ea   bridge    bridge    local
e9d223c049ad   host      host      local
76287cca2804   none      null      local
```
Looks like `docker endpoint for "default" not found` means no default network can be created

**Comment by QuAzI** [comment]
Minimal config for `dotnet new webapp` which also throws problem
[dotnet new webapp.zip](https://github.com/docker/compose/files/10688065/dotnet.new.webapp.zip)

**Comment by yilmazdurmaz** [comment]
Ok, hold on again but this time for a misunderstanding. Your details have a few extras to look for such as using both Windows and Linux containers, and Ubuntu 22.04.

But before going on, the folder you should be removing is `C:\Users\<yourusername>\.docker\contexts\meta\etc..`, not the one inside the WSL. Can you check, with this correction, if you will get the problem this frequently?

**Comment by QuAzI** [comment]
Interesting... After I removed `%USERPROFILE%\.docker\contexts\meta` my last minimal test with webapp works fine. But previous test creates empty meta.json again
[problematic docker-files.zip](https://github.com/docker/compose/files/10688289/docker-files.zip)

**Comment by nelsonjchen** [comment]
https://serverfault.com/questions/336630/is-there-a-quick-way-to-see-what-filesystem-filter-drivers-are-loaded-on-windows

I'm curious what the output here is. To be honest, I'm no Windows driver developer but this does seem interesting.

**Comment by QuAzI** [comment]
@nelsonjchen 
```
C:\WINDOWS\system32>fltmc filters

Filter Name                     Num Instances    Altitude    Frame
------------------------------  -------------  ------------  -----
bindflt                                 1       409800         0
FsDepends                              11       407000         0
DwProt                                 11       323610         0
SpiderG3                               11       323600         0
storqosflt                              2       244000         0
wcifs                                   3       189900         0
CldFlt                                  6       180451         0
FileCrypt                               0       141100         0
luafv                                   1       135000         0
npsvctrig                               1        46000         0
Wof                                     8        40700         0
FileInfo                               11        40500         0

C:\WINDOWS\system32>fltmc volumes
Dos Name                        Volume Name                              FileSystem   Status
------------------------------  ---------------------------------------  ----------  --------
                                \Device\Mup                              Remote
D:                              \Device\HarddiskVolume7                  NTFS
C:                              \Device\HarddiskVolume5                  NTFS
                                \Device\NamedPipe                        NamedPipe
                                \Device\Mailslot                         Mailslot
                                \Device\HarddiskVolume3                  FAT
E:                              \Device\HarddiskVolume2                  NTFS
                                \Device\HarddiskVolume6                  NTFS
C:\ProgramData\Microsoft\Windows\Containers\BaseImages\2fd8febc-ccab-468c-a514-3c07da17377e\BaseLayer  \Device\HarddiskVolume10                 NTFS
K:                              \Device\HarddiskVolume33                 FAT
                                \Device\HarddiskVolumeShadowCopy3        NTFS
                                \Device\HarddiskVolumeShadowCopy4        NTFS
                                \Device\HarddiskVolumeShadowCopy5        NTFS
```

**Comment by yilmazdurmaz** [comment]
@QuAzI , I tested against having a multi-layer in the build file. I modified it for a sample ASP.NET application. both the build file and compose files worked fine for me. That suggests something else in your system (docker service?) retains that empty meta file and rewrites it back into contexts. try "down" first then "up" again containers created with compose, clearing older images possibly created during your struggles, and restarting the host machine.

**Comment by cforce** [comment]
Delete the meta.json file %USER_PROFILE%\.docker\contexts\meta\{some ver random characters }\meta.json - afterwards it worked again

**Comment by Zetanova** [comment]
deleting the file under `%USER_PROFILE%\.docker\contexts\meta{some ver random characters }\meta.json` 
then start docker host and run `docker system prune` twice fixes the problem more persistent.
Its some kind of a broken image layer that will trigger this timeout on a specific container startup

I could start mssql, eventstore, seq,
2 of my visual studio docker-compose projects without issue
and a 3th  visual studio docker-compose project docker host simply timed out
visual studio life locked. 

Only to execute `docker system prune` twice fixes the problem.

**Comment by milas** [comment]
Temporarily locking this because we're getting a lot of redundant comments, and I know a lot of people are subscribed here so want to avoid excessive notifications for everyone! 🙃 

The team is working on this internally, and we'll continue to share updates here.

For now, if you encounter this error while running `docker` or `docker compose` commands and you have an empty `meta.json` file in your Docker contexts, you'll need to manually remove it.

On Windows, `meta.json` files can be found in subdirectories of `%USERPROFILE%\.docker\contexts\meta`.

**Comment by milas** [comment]
Docker Desktop 4.17 has been released and includes a fix for this 🥳 

We do a staged rollout of new releases, so you might not receive a prompt for the in-app update immediately, but you can always click "Check for Updates" in settings or [download the latest version manually](https://docs.docker.com/get-docker/)

Thanks for everyone's patience, bug reports, and sharing of workarounds in the meantime!

**Comment by yilmazdurmaz** [comment]
@ntx-ben, have you followed the basic steps? please first test these and report back.
1. what is the output to `docker context ls` command? is the one in the title of this topic? do you see at least "default *" in the output? else copy-paste the output here.
2. check meta files under this folder: `%USERPROFILE%\.docker\contexts\meta`. and then remove anything in there (make a backup if you have custom contexts).

**Comment by vwvm** [comment]
> This solution successfully solved my problem
delete C:\Users\~\.docker

**Comment by vwvm** [comment]
This solution successfully solved my problem
delete C:/user/~/.docker

**Comment by vwvm** [comment]
This solution successfully solved my problem
delete C:/user/~/.docker

**Comment by maietta** [comment]
> solved by: reinstalling docker desktop version 4.12.0

What the hell is wrong with you people? That's like me telling you to re-install Windows when a program crashes. This is NOT a solution.

**Comment by maietta** [comment]
> @ntx-ben, have you followed the basic steps? please first test these and report back.
> 
> 1. what is the output to `docker context ls` command? is the one in the title of this topic? do you see at least "default *" in the output? else copy-paste the output here.
> 2. check meta files under this folder: `%USERPROFILE%\.docker\contexts\meta`. and then remove anything in there (make a backup if you have custom contexts).

$ docker context ls
unexpected end of JSON input

**Comment by yilmazdurmaz** [comment]
Hi @maietta , Reinstalling sometimes solves problems, but only sometimes. So it still keeps staying in possible solutions, and worth to give a shot.

Anyways, as you noticed, the problem is those broken meta files that exist even after an update/reinstall process. Developers put lots of effort into it and this should not occur in 4.17 and onwards.

However, it is still possible the update cannot clean up all currently broken files, so the main solution is to find broken context files under `%USERPROFILE%\.docker\contexts\meta` and remove them or replace them from the backup if any. 

Good Luck ;)

**Comment by milas** [comment]
Docker Desktop 4.17+ has fixes around this. If you see it after upgrading, manually clean the file (as mentioned ⬆️ @ `%USERPROFILE%\.docker\contexts\meta`).

**Comment by yilmazdurmaz** [comment]
Hi again @ntx-ben,
press <kbd>Windows</kbd> key + <kbd>R</kbd>, type "cmd" and press <kbd>Enter</kbd> (any method if you use frequently) and run this command to rename profile folder, then try restarting docker desktop.
````
move %USERPROFILE%\.docker %USERPROFILE%\.dockerx
````
if this works, it means something else in this folder was broken. if so, keep the folder for further analysis to help develop a fix but do not share with anyone until you are sure you do not leak private info from within.

if not, try this link to see service logs: https://docs.docker.com/config/daemon/logs/
logs are under `%USERPROFILE%\AppData\Local\Docker\log\vm` and use `%USERPROFILE%\.docker\daemon.json` to change/enable debug levels.
- before trying to start, clear all log files from that log folder to purge results from previous attempts. makes inspection easier.
- if no JSON pro, just append `"debug": true,` (ends with comma) after first open curly brace `{` (`log-level` also might be needed to see not the errors but also when it gets stuck)

see if you can figure it out from the logs, else check them for private information, redact them and share logs with us for extra pair of eyes.

**Comment by ntx-ben** [comment]
Ok, it turns out I was running WSL 2 with a custom kernel to debug `Cilium`, which somehow became incompatible with Docker? 

So I remove the custom kernel from the `.wslconfig` file, restarted WSL and now I am able to use Docker Desktop `v4.17.1`

Thanks for the assistance. Case closed.

**Comment by dallenbaldwin** [comment]
I keep running into this issue. it's sporadic so it's hard to know what's causing it. But I do have a question that might be relevant.

If I'm running docker through WSL, should there be a .docker folder in my Windows user directory and WSL user directory? Could it be possible that the two are fighting and causing issues?

**Comment by yilmazdurmaz** [comment]
Hi @dallenbaldwin, The issue was solved since v4.17, but it is possible someone broke the fix.
Please first check a few things:
- run `docker context ls` first, check the discussion for the resulting error.
- check the last post @ntx-ben gave just before your post.

Check the answers here, specificaly my messages as I tried to summarize possible solutions as much as possible. The discussion is long, but keep your patience as it may possibly solve your issue.

If all fails, we may have a new issue at hand. Then
- Open a new bug post
- enter as much detail as required; mainly versions of everything related
- also give a link back to this post so we can follow it without losing its tracks

**Comment by harmancode** [comment]
I'd just like to note here that upgrading to **Docker Desktop 4.22.1 (118664)** solved this issue for me on Ubuntu 20.04.5 LTS on WSL 2 with Windows 10. 

My previous Docker versions: 
```
Docker version 20.10.12, build 20.10.12-0ubuntu2~20.04.1
Docker Compose version v2.15.1
```

**Comment by yilmazdurmaz** [comment]
@Philly-B 

> docker context ls open >somepath>

this part might be the issue if you used the command as you wrote here. there is no "open" in its parameter list. maybe you copied wrong parts from some tutorial.

try again and use only `docker context ls`. check the error again and see if it is the same as in the title of this discussion. your problem might not be what we discussed here. check the existence and content of meta files (if broken) as we discussed here.

for what you said about docker machine and WSL2; while you install Docker Desktop, it asks you if you want WSL, and then it takes care of "machine" during the installation. so, in a way, it is still there, but not as a virtual machine you may think of.

**Comment by Philly-B** [comment]
@yilmazdurmaz 

Sadly it is the same when I execute `docker context ls` . Also the first error I see when I start Docker is `Process com.docker.dev-envs.exe has exited`. Sometimes that did not show and then the first error was the issue described here. I also deleted and reinstalled (with cleaning the env variables) multiple times and also deleted the .docker folder. 

As of now it seems like the main problem is that docker does not create the machine folder within the .docker folder. Also when Docker desktop is started I can see in the bottom left that the engine is running, but when I go the settings / builders I can see that the default has an error: `Get "http://192.168.99.100:2376/_ping": context deadline exceeded`

It is really confusing to me that there is no clear error here. Probably the only consistent thing is that somehow docker can not connect to the "machine".

**Comment by Philly-B** [comment]
For all the fellow docker users on windows 11 with the same issue. I could solve it by switching the docker context to "desktop-linux".

At least for me, there were two entries for `docker context ls:`

`default *       Current DOCKER_HOST based configuration   tcp://127.0.0.1:2376`
`desktop-linux   Docker Desktop                            npipe:////./pipe/dockerDesktopLinuxEngine`

I removed the environment variable `DOCKER_HOST` and switched executed ` docker context use desktop-linux` . Now it magically works. I hope it outlives a pc restart.

**Comment by yilmazdurmaz** [comment]
@Philly-B  thanks for sharing a solution.

the issue might be ``DOCKER_HOST`` this whole time. I haven't used it before so I will just speculate that having it on *default* context is not desirable 😅  can you check the default one again with this variable deleted? (it seems it should be used to access a remote daemon)

PS: as long as you continue using that new context, or any non-broken context, it should be fine.

**Comment by Philly-B** [comment]
@yilmazdurmaz 

I switched the context back:

`PS C:\Users\Phil> docker context ls
NAME            DESCRIPTION                               DOCKER ENDPOINT                             ERROR
default *       Current DOCKER_HOST based configuration   tcp://127.0.0.1:2376
desktop-linux   Docker Desktop                            npipe:////./pipe/dockerDesktopLinuxEngine`

Restarted the docker engine but still get an error on any docker command:

`PS C:\Users\Phil> docker ps -a
error during connect: Get "https://127.0.0.1:2376/v1.47/containers/json?all=1": dial tcp 127.0.0.1:2376: connectex: No connection could be made because the target machine actively refused it.`

**Comment by yilmazdurmaz** [comment]
@Philly-B  this "connection refused" part ... to me, that last message suggests, rather than broken context, that your issue might be from "forgotten/broken" environment variables. check other ``DOCKER_*`` variables in your system, TLS for example. Port 2376 is for secure connections, and maybe either some variables are missing, or they exists but shouldn't.

up for some more testing?

**Comment by Philly-B** [comment]
Sorry for the late reply. I actually got a new computer for Christmas, which did not have the problem. I "sadly" can't reproduce it anymore.
